py/dml/structure.py

```diff
@@ -77,7 +77,7 @@ def mkglobals(stmts):
     by_name = {}
     assert not global_scope.symbols()
     for stmt in stmts:
-        if stmt.kind in ['extern', 'extern_typedef', 'dml_typedef']:
+        if stmt.kind in ('extern', 'extern_typedef', 'dml_typedef'):
```

Comment:
A set literal is actually the optimal choice these days, also for small sets.

Reply:
Really! Is Python actually smart enough to let-float it out (err, I suppose _JIT_-float it out?) and make the check O(1)?

Reply:
I don't know, all I know is that tuples were better in py2 but sets are better in py3:
```
>>> timeit.timeit("'foo' in {'extern', 'extern_typedef', 'dml_typedef'}")
0.015403441153466702
>>> timeit.timeit("'foo' in ('extern', 'extern_typedef', 'dml_typedef')")
0.020750856958329678
```

Reply:
jesus how does this take 2 hundreths of a second

python slow

Reply:
Oh wait, no, `timeit` is designed for small code snippets. It's not the clown! _I_ am the clown! Oh no!

Reply:
Refactored all literals used for membership checking to be set literals